src/ui/common/src/components/pages/workflow/id/hook.ts

```diff
@@ -20,11 +28,58 @@ export type useWorkflowOutputs = {
   workflowDagResultWithLoadingStatus: WorkflowDagResultWithLoadingStatus;
 };
 
+// useWorkflowIds ensures we use the URL parameter as ground-truth for fetching
+// workflow, dag, and result IDs. It includes additional hooks to ensure
+// redux states are in-sync.
+// This hook should be used for all pages that need to access a single DAG (or DAG result)
+// data.
+export function useWorkflowIds(apiKey: string): useWorkflowIdsOutputs {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const {
+    id: wfIdParam,
+    dagId: dagIdParam,
+    dagResultId: dagResultIdParam,
+  } = useParams();
+
+  const { data: dagResults } = useDagResultsGetQuery(
+    { apiKey, workflowId: wfIdParam },
+    { skip: !wfIdParam }
+  );
+
+  // Select the first availale dag result if ID is not provided.
+  const dagResult = dagResultIdParam
+    ? (dagResults ?? []).filter((r) => r.id === dagResultIdParam)[0]
```

Comment:
Can you add a comment about what `(dagResults ?? []).filter((r) => r.id === dagResultIdParam)[0]` does?
I think it's saying find the `dagResult` from `dagResults` that has an id that matches `dagResultIdParam` but I am not too sure what happens if there is none. I think it would access the 0-th item of an empty list so it might error out.

Reply:
`dagResults ?? []` ensures we always operates on a list even if `dagResults` is undefined, and `[0]` will return an undefined object if the list is empty (typescript generally gives undefined if you try to access an out-of-bound object in array). In this way the availability is reflected on whether `dagResult` is `undefined`.